docs/EngineVsOperation.md

```diff
@@ -0,0 +1,10 @@
+# Engine versus Operation  
+
+During the creation of this implementation it was decided to break the implementation into two parts.  The first  is the individual operations that perform the data transforms. The operations are handles as specific functions that are contributed individually through a given issue request to match the specification.  Each Operation is constructed of at least the following:
+
+    * operation.go and operation_test.go files that define the function the operation performs and its tests.
+    * metadata.go which defines the inputs and parameters of the operation
+    * a README.md that describes how close the operation implementation is the the specification
+    * a descriptor.json that defines additional data about the operation such as the version of specification the operation complies with.
+
+The second portion of this implementaiton is the Engine that reads the structure JSON, handles the data in memory, gets inputs, writes outputs, and defines a operation object that is then extended for the individual operations.  The engine consists of a [flogo action](https://github.com/project-flogo/core/blob/master/action/action.go) that handles a given event a data to be manipulated making it a strong tool for this implementation.
```

Comment:
Should we rename engine to something else? Because the thing you are talking about is actually `flogo-action` and just adding engine notation to it will create confusion with `flogo-engine`

Reply:
but assuming someone knows what an action is is just confusing for the people these documents are intended for.  The action is basically the engine that runs everything in the back ground.  Engine is a generic term, flogo-engine is something specific.  I agree it is confusing but I am trying to address a general developer not a flogo dev.  I didn't even know there was a flogo-engine object.

Reply:
I think, pipeline should be a better term based on what it's doing. This should be clear to both flogo-dev and non flogo-dev. What do you think ?

Reply:
pipeline or structure is what a given json is (the chain of operations is a pipeline)

Reply:
I talked to Matt and he thought engine was fine, and wouldn't be confusing.